clients/src/main/java/org/apache/kafka/common/TopicCollection.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common;
+
+import java.util.Collection;
+
+/**
+ * A class used to represent a collection of topics. This collection may define topics by topic name
+ * or topic ID. Subclassing this class beyond the classes provided here is not supported.
+ */
+public abstract class TopicCollection {
```

Comment:
Would it be helpful to have a couple static factories? For example:
```java
public static TopicIdCollection ofTopicIds(Collection<Uuid> uuids);
public static TopicNameCollection ofTopicNames(Collection<String> names);
```

Reply:
Static factories in addition to a public constructor? Or make the constructor private?

Reply:
No strong opinion, but I'd probably vote to keep the constructors private. Might be worth getting a second opinion. @ijuma ?

Reply:
I think it's simpler with just the factories, so I like this idea.